ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/UnnecessaryParenthesesRule.kt

```diff
@@ -0,0 +1,41 @@
+package com.pinterest.ktlint.ruleset.experimental
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.LAMBDA_ARGUMENT
+import com.pinterest.ktlint.core.ast.ElementType.LPAR
+import com.pinterest.ktlint.core.ast.ElementType.RPAR
+import com.pinterest.ktlint.core.ast.ElementType.VALUE_ARGUMENT_LIST
+import com.pinterest.ktlint.core.ast.children
+import com.pinterest.ktlint.core.ast.isPartOf
+import com.pinterest.ktlint.core.ast.nextCodeSibling
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+class UnnecessaryParenthesesRule : Rule("unnecessary-parentheses") {
```

Comment:
Rule has been renamed to "unnecessary-parentheses-before-trailing-lambda".

Out of curiosity, in what what way would "we" need to implement the same behavior in detekt? Is there some kind of collaboration with that project?

Reply:
We don't necessarily need to implement the **same** behavior as in detekt, I just meant that the name should be more specific because before you changed it, as a user of the tool, I'd expect it to do more than just parens before trailing lambdas.